Aminophen cjk

norbusan opened the pull request:
pull request 7 moved here:
By aminophen:
これまで texjporg/ckfontmaps で作ってきた中国語・韓国語の map を、すべて jfontmaps に入れるという提案です。同時に、repository の名前を jfontmaps ではなく ptex-fontmaps に変えることを提案します。

今のところ、release.sh は

    これまであった日本語は free + nonfree
    新しい中国語・韓国語は free だけ

を CTAN に upload するように書いてあります。

norbusan committed:
update release.sh for free/nonfree split

norbusan committed:
update README for nonfree, add README.nonfree

norbusan committed:
Merge branch 'master' into split-nonfree

aminophen committed:
Initial commit

aminophen committed:
Add korean maps

aminophen committed:
Add simplified chinese maps

aminophen committed:
Add traditional chinese maps

aminophen committed:
README: tweaked

aminophen committed:
Add more maps

aminophen committed:
Add more maps

aminophen committed:
Add solaris map (korean)

aminophen committed:
Import updmap.pl and updmap.cfg from TeX Live svn r42638

aminophen committed:
updmap.pl, updmap.cfg: Add support for ckfontmaps

norbusan committed:
simplify replacement code

norbusan committed:
white space fixing

aminophen committed:
Add more maps (cjk-unifonts, ttc/ttf edition)

aminophen committed:
Add PSName to various maps

aminophen committed:
Patches for TeX Live source

aminophen committed:
Merge remote-tracking branch 'origin/split-nonfree'

aminophen committed:
Update README, release.sh

aminophen committed:
README: current status

aminophen committed:
README: current status (typo)

aminophen committed:
Import Chinese/Korean maps

norbusan committed:
Revert "update README for nonfree, add README.nonfree"

This reverts commit e0e833f0f130c32d77a34e9489092e865ebd5c16.

norbusan committed:
Revert "update release.sh for free/nonfree split"

This reverts commit 684abfb0719673da3c4e72722648222f7a4cdd40.

norbusan commented:
とりあえずsplit nonfreeのコミットをrevertしました。

aminophen commented:
#1 にコメントしました。

aminophen commented:
split-nonfree を revert していただいているので、merge しても大丈夫でしょうか？

norbusan commented:
TODO：

- 名前はどうしますか？「ptex-fontmaps」に変更していいですか？
- `scEmbed`, `tcEmbed`, `koEmbed`みたいに`jaEmbed`にして、`kanjiEmbed`のaliasとして
- `kanji-config-updmap`の対応

norbusan commented:
mergeをもう少し待ってください。上のTODOをやりたいです。

aminophen commented:
> 上のTODOをやりたいです。

了解です。

norbusan commented:
名前の変更は大丈夫ですか？他の開発者は？

norbusan committed:
use svn-id free code for development

norbusan committed:
switch from kanji* to ja*

norbusan committed:
tl-update, use jaEmbed, jaVariant for font maps

norbusan committed:
update release for renaming jfontmaps -> ptex-fontmaps

norbusan committed:
adjust kanji-config-updmap.pl for ja/tc/sc/ko

norbusan commented:
上のTODOを全部やりましたが、testingはまだまだです。いろいろな変更があったので、たくさんのtestingしないといけません。
これから、`kanjiEmbed`じゃなくて、`jaEmbed` `jaVariant`を使うことになりますが、古い名前をまだ使えます。

aminophen committed:
kanji-config-updmap.pl: typo

aminophen commented:
テストしてみました。kanji-config-updmap.pl の representatives が間違っていたので直しました。ほかは ok のように見えます。

norbusan commented:
ありがとう。updmapをもう試してみましたか？

aminophen commented:
`kanji-config-updmap --ko apple` や `kanji-config-updmap status --mode=sc` が正しく動くので、updmap も問題ないと思いました。kanjiEmbed から jaEmbed への backward compatibility も正しく動いていると思います。

norbusan commented:
ありがとうございます、すごく助かります！

aminophen commented:
よくみると kanjiEmbed → jaEmbed の変更で WARNING が出ていました。

~~~~
updmap [WARNING]: unrecognized line 0 in /usr/local/texlive/2016/texmf-config/web2c/updmap.cfg: kanjiEmbed hiragino-pron
updmap [WARNING]: unrecognized line 1 in /usr/local/texlive/2016/texmf-config/web2c/updmap.cfg: kanjiVariant 
~~~~

updmap.pl から `get_cfg('kanjiEmbed')` と `get_cfg('kanjiVariant')` がなくなったからでしょうか？

norbusan committed:
more backward compatibility fixes

norbusan commented:
fixをプッシュしました。

aminophen commented:
ありがとうございます。いままでの kanjiEmbed / kanjiVariant の設定を引き継げるようになったのを確認しました。ok だと思います。

aminophen committed:
README: typo

aminophen committed:
README: typo

aminophen committed:
updmap.pl: typo

aminophen commented:
Simplified Chinese の sinotype は、なぜか実際に使ってみると失敗しました。どうやらこれは Chinese TeX Society も問題を知っているようです。ctex.pdf より：

> 在DVIPDFMx 下，可以通过下述方式使用华文宋体和华文楷体：
> ~~~~
>   \special{pdf:mapline unisong@Unicode@ unicode :4:Songti.ttc}
>   \special{pdf:mapline unikai@Unicode@ unicode :4:Kaiti.ttc}
> ~~~~
> 而\pdfmapline 似乎不支持带索引的ttc 字体，Songti.ttc 默认使用的是Songti SC Black，Kaiti.ttc 默认使用的是Kaiti SC Black。华文黑体不能通过这种方式使用：
> ~~~~
>   \special{pdf:mapline unihei@Unicode@ unicode \detokenize{华文黑体}.ttf}
> ~~~~
> DVIPDFMx 不能生成PDF，报下述错误：
> ~~~~
>   ** WARNING ** UCS-4 TrueType cmap table...
>   ** ERROR ** Unable to read OpenType/TrueType Unicode cmap table.
> ~~~~
> 如果将CMap 改为UniGB-UTF16-H，错误信息是
> ~~~~
>   ** WARNING ** No usable TrueType cmap table found for font "华文黑体.ttf".
>   ** WARNING ** CID character collection for this font is set to "Adobe-GB1"
>   ** ERROR ** Cannot continue without this...
> ~~~~
> 在pdfTEX 下生成的PDF 只有方框。华文细黑和华文仿宋的情况类似。这确认为DVIPDFMx的bug。

dvipdfmx のバグ？とのことなので、とりあえず sinotype の map は disable したほうがよいかもしれないと思います。もう少し調べます。

----

dvipdfmx のバグならばそれを直したほうがよいと思いますが、Mac OS 10.7 に付いている nonfree フォントなので、再現が難しいかもしれません。具体的には dvipdfmx の cidtype2.c に書いてある

https://github.com/clerkma/ptex-ng/blob/master/texk/libdpx/cidtype2.c#L597

に引っかかっている可能性があります。

aminophen commented:
よくみると dvipdfmx bug ではなさそうでした。STSong.ttf と STHeiti.ttf の cmap を parse してみると、Adobe-GB1 の cmap は入っていませんでした。

~~~~
upstsl-h	unicode	STSong.ttf
upstsl-v	unicode	STSong.ttf
upstht-h	unicode	STHeiti.ttf
upstht-v	unicode	STHeiti.ttf
~~~~

なら ok です（ただし、STHeiti.tttf は問題ありませんが、STSong.ttf のほうは
~~~~
dvipdfmx:warning: Embedding of font "STSong.ttf" disabled due to license restrictions
~~~~
が出て非埋め込みになるので、dvipdfmx -E test.dvi のようにすると license restrictions を無視して埋め込む）。ということは sinotype は Adobe-GB1 の仕組みで呼び出すのは無理ですから、私たちが作ろうとしている uplatex / otf 向け map files には向かないということです。というわけで削除します。

aminophen committed:
remove sinotype for simplified chinese

aminophen commented:
今度はなぜか Simplified Chinese の noEmbed だけがうまく動いていないことに気づきました。

~~~~ tex
\documentclass[uplatex]{jsarticle}
\usepackage[english]{babel}
\usepackage[main=japanese]{pxbabel}
\begin{document}
\foreignlanguage{schinese}{早上好，谢谢你。}\par
\foreignlanguage{schinese}{\textgt{早上好，谢谢你。}}\par
\foreignlanguage{tchinese}{早上好，謝謝你。}\par
\foreignlanguage{tchinese}{\textgt{早上好，謝謝你。}}\par
\foreignlanguage{korean}{안녕, 감사합니다.}\par
\foreignlanguage{korean}{\textgt{안녕, 감사합니다.}}\par
\end{document}
~~~~
~~~~
$ uplatex test.tex
$ dvipdfmx dvipdfmx -f otf-ko-noEmbed.map -f otf-sc-noEmbed.map -f otf-tc-noEmbed.map -f uptex-ko-noEmbed.map -f uptex-sc-noEmbed.map -f uptex-tc-noEmbed.map test.dvi
~~~~

これを実行すると
~~~~
dvipdfmx:warning: Could not locate a virtual/physical font for TFM "upstht-h".
dvipdfmx:warning: >> This font is mapped to a physical font "STHeiti-Regular".
dvipdfmx:warning: >> Please check if kpathsea library can find this font: STHeiti-Regular
dvipdfmx:fatal: Cannot proceed without .vf or "physical" font for PDF output...
~~~~

でエラーが出て PDF が出ません。

ソースから `\foreignlanguage{schinese}{\textgt{早上好，谢谢你。}}\par` を削除すると

~~~~
dvipdfmx:warning: CMap has higher supplement number than CIDFont: STSong-Light
dvipdfmx:warning: Some chracters may not be displayed or printed.
dvipdfmx:warning: CMap has higher supplement number than CIDFont: MSung-Light
dvipdfmx:warning: Some chracters may not be displayed or printed.
dvipdfmx:warning: CMap has higher supplement number than CIDFont: MHei-Medium
dvipdfmx:warning: Some chracters may not be displayed or printed.
~~~~

が出ますが、PDF が出てくるので、さっぱりわかっていません。

aminophen committed:
Add fandol family (sc)

aminophen committed:
README: fandol is shipped with TeX Live

aminophen commented:
> なぜ STHeiti-Regular だけうまくいかないのか

[forum:2110](http://oku.edu.mie-u.ac.jp/tex/mod/forum/discuss.php?d=2110) を経て [r42803](http://www.tug.org/svn/texlive?view=revision&revision=42803) で ok になりました。これで全ての map および updmap と kanji-xonfig-updmap スクリプトの動作確認がとれました。

qinglee committed:
Prefer fandol to arphic

Both of them are shipped with TeX Live. But fandol fonts are higher quality than araphic.

norbusan commented:
山下さん、あけましておめでとうございます。現在の状況はそのまま大丈夫でしょうか？
最初にupdmapの更新しないと行けないと思います。その後ptex-fontsmapsをアップして、jfontmapsのかわりにTeX Liveに入るつもり。
この計画はよろしいでしょうか。

aminophen commented:
ノルベルトさん

あけましておめでとうございます。

merge していただいて大丈夫です。計画も良いと思います。よろしくお願いします。

aminophen commented:
私からのお願いは

- 「ptex」なので、collection-langcjk ではなく collection-langjapanese に入ってほしい
- CTAN の uploader info entry は `Japanese TeX Development Community <issue@texjp.org>` にしてほしい（そうすれば texjporg member なら誰でも upload できます）

あと、もし日本人が c-langjapanese だけインストールした場合、この updmap-hdr.cfg の default fonts が見つからないと思うので、その場合にエラーが出ないかだけ確認してほしいです。

もしかすると ptex-fontmaps の depend に
~~~~
depend arphic-ttf
depend fandol
depend baekmuk
~~~~
があるとよい？

norbusan commented:
Hmmm, `tcEmbed`,などのデフォルトは`noEmbed`にすればどうでしょうか？

norbusan committed:
updmap.pl replace also @kanji*@ strings for backward compatibility

aminophen commented:
> Hmmm, tcEmbed,などのデフォルトはnoEmbedにすればどうでしょうか？

それが[このコメント](https://github.com/texjporg/jfontmaps/pull/8?_pjax=%23js-repo-pjax-container#issuecomment-269598998)にあるように、[r42803](http://www.tug.org/svn/texlive?view=revision&revision=42803) より新しい dvipdfmx でないと `scEmbed noEmbed` が使えないという問題があります。

norbusan commented:
なるほど。ありがとう。考えています。。。

norbusan committed:
add a ptex-fontmaps.tlpsrc with necessary font depends

aminophen committed:
kanji-config-updmap.pl: representative batang.ttf -> dotum.ttf

aminophen commented:
updmap.pl を読みましたが、actual font files がなくても maps があれば updmap.pl は失敗しないですね。TeX Live の post_install でエラーが起きる心配はしなくてよかったようです。

よって、中国語・韓国語のフォントは depend しなくてもよいかもしれません。中国語と韓国語が必要な人だけ、自分でインストールすればよいと考えればそれでもよい気がしてきました。

aminophen committed:
scEmbed default: fandol -> arphic-ttf
This reverts the commit 142decd, for smaller dependency

aminophen commented:
とりあえず simplified chinese を arphic にすると depend の数が減るので、減らしておきました。

長い間 aminophen-cjk で作ってきましたが、merge してよいでしょうか。

norbusan committed:
don't put tl-update dir into release tarball

norbusan commented:
merge とuploadしました。

aminophen commented:
ありがとうございます！
